go/protocols/flow/re_exports.go

```diff
@@ -70,6 +71,52 @@ func RunAsyncOperation(fn func() error) OpFuture {
 	return op
 }
 
+type OpFutureWithState interface {
+	OpFuture
+	// State emitted by OpFuture to be sent as part of the Acknowledge response
+	State() *ConnectorState
+}
+
+type AsyncOperationWithState struct {
+	op    *AsyncOperation
+	state *ConnectorState
+}
+
+func (o AsyncOperationWithState) Done() <-chan struct{} {
+	return o.op.Done()
+}
+
+func (o AsyncOperationWithState) Err() error {
+	return o.op.Err()
+}
+
+func (o AsyncOperationWithState) State() *ConnectorState {
+	return o.state
+}
+
+func RunAsyncOperationWithState(fn func() (*ConnectorState, error)) OpFutureWithState {
+	var op = NewAsyncOperation()
+	// This is considered a no-op connector state
+	var s = ConnectorState{}
```

Comment:
Is this definitely considered a no-op? I don't want to accidentally wipe checkpoints by this 👀